keps/sig-network/1880-multiple-service-cidrs/README.md

```diff
@@ -0,0 +1,954 @@
+ranges, for consistency, but there should not be any functional problem in the cluster if the
+Service implementation (kube-proxy, ...) is able to handle those IPs.
+
+Example:
+
+- flags set to 10.0.0.0/20
+- upgrade to 1.25 with alpha gate
+- apiserver create a default ServiceCIDRConfig object for 10.0.0.0/20
+- user creates a new ServiceCIDR for 192.168.1.0/24
+- create a Service which gets 192.168.1.1
+- rollback or disable the gate
+- the apiserver repair loops will generate periodic events informing the user that the Service with the IP allocated
+  is not within the configured range
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: ServiceCIDRConfig
+  - Components depending on the feature gate: kube-apiserver, kube-controller-manager
+
+###### Does enabling the feature change any default behavior?
+
+The time to reuse an already allocated ClusterIP to a Service will be longer, since the ClusterIPs
+now depend on an IPAddress object that is protected by a finalizer.
+
+The bootstrap logic has been updated to deal with the problem of multiple apiservers with different
+configurations, making it more flexible and resilient.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+If the feature is disabled the old allocator logic will reconcile all the current allocated IPs
+based on the Services existing on the cluster.
+
+If there are IPAddresses allocated outside of the configured Service IP Ranges in the apiserver flags,
+the apiserver will generate events referencing the Services using IPs outside of the configured range.
+The user must delete and recreate these Services to obtain new IPs within the range.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+There are controllers that will reconcile the state based on the current created Services,
+restoring the cluster to a working state.
+
+###### Are there any tests for feature enablement/disablement?
+
+<!--
+The e2e framework does not currently support enabling or disabling feature
+gates. However, unit tests in each component dealing with managing data, created
+with and without the feature, are necessary. At the very least, think about
+conversion tests if API types are being modified.
+
+Additionally, for features that are introducing a new API field, unit tests that
+are exercising the `switch` of feature gate itself (what happens if I disable a
+feature gate after having objects written with the new field) are also critical.
+You can take a look at one potential example of such test in:
+https://github.com/kubernetes/kubernetes/pull/97058/files#diff-7826f7adbc1996a05ab52e3f5f02429e94b68ce6bce0dc534d1be636154fded3R246-R282
+-->
+
```

Comment:
We definitely will need these tests. You should probably include a test plan here or reference one.

Reply:
I thought these questions were only for beta, but this should be doable to do with the integration framwork

Reply:
added a comment , I'll fill the test sections during the implementation, but I expect to spend as much time on testing as on feature development

Reply:
enable/disable is for alpha, later questions are for beta and GA

Reply:
yeah, I'll add the test, I'm specially interested in this and the skewed problem,